Applications/ConsoleReferenceServer/Program.cs

```diff
@@ -143,33 +159,28 @@ public class MyRefServer
         private ReferenceServer m_server;
         private Task m_status;
         private DateTime m_lastEventTime;
-        private bool m_autoAccept = false;
-        private bool m_logConsole = false;
-        private static ExitCode s_exitCode;
+        public bool LogConsole { get; set; } = false;
+        public bool AutoAccept { get; set; } = false;
+        public string Password { get; set; } = null;
+        public ExitCode ExitCode { get; private set; }
 
-        public MyRefServer(bool autoAccept, bool logConsole)
-        {
-            m_autoAccept = autoAccept;
-            m_logConsole = logConsole;
-        }
-
-        public void Run()
+        public async Task Run()
         {
             try
             {
-                s_exitCode = ExitCode.ErrorServerNotStarted;
-                ConsoleSampleServer().Wait();
+                ExitCode = ExitCode.ErrorServerNotStarted;
+                await ConsoleSampleServer().ConfigureAwait(false);
```

Comment:
Async method should end with "Async". Also, name does not indicate what method does. Suggestion: StartSampleConsoleServerAsync

Reply:
StartConsoleReferenceServerAsync ...